pkg/controller/core/workload_controller.go

```diff
@@ -169,10 +169,12 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			// The deactivation reason could be deduced as the maximum number of re-queuing retries if the workload met all criteria below:
 			//  1. The waitForPodsReady feature is enabled, which means that it has "PodsReady" condition.
 			//  2. The workload has already exceeded the PodsReadyTimeout.
-			//  3. The number of re-queued has already reached the waitForPodsReady.requeuingBackoffLimitCount.
+			//  3. The workload already has been re-queued previously, which means the requeueAt has already passed.
+			//  4. The number of re-queued has already reached the waitForPodsReady.requeuingBackoffLimitCount.
 			if apimeta.IsStatusConditionFalse(wl.Status.Conditions, kueue.WorkloadPodsReady) &&
 				((!workload.HasRequeueState(&wl) && ptr.Equal(r.requeuingBackoffLimitCount, ptr.To[int32](0))) ||
-					(workload.HasRequeueState(&wl) && ptr.Equal(wl.Status.RequeueState.Count, r.requeuingBackoffLimitCount))) {
+					(workload.HasRequeueState(&wl) && wl.Status.RequeueState.RequeueAt != nil && wl.Status.RequeueState.RequeueAt.Sub(realClock.Now().Truncate(time.Second)) <= 0 &&
```

Comment:
What is the reason to check the `requeueAt`? The field should be set to nil rather than compared. It is set to nil on the main branch.

Reply:
> The field should be set to nil rather than compared. 

No, in the release 0.6, the `requeueAt` wouldn't reset to nil. That is functionality only for the main branch.

Reply:
I would argue that cleaning `requeueAt` is not a functionality on main, but a bugfix.

I think keeping the `requeueAt` when backoffLimitCount is exceeded might be misleading to the user. I would rather suggest to investigate clearing `requeueAt` in 0.6 as we do on the `main` branch, when the backoffLimitCount is exceeded.

Reply:
I synced with @mimowo offline. In conclusion, we should cherry-pick https://github.com/kubernetes-sigs/kueue/pull/2143 so that we can bring the functionality to reset the `requeueAt` in v0.6 as well.

By doing so, we can avoid the current hacky mechanism.
But, we can not just cherry-pick #2143 to v0.6 since #2143 depends on the `Requeued` condition, which is the condition type only in v0.6.

Reply:
I opened the issue https://github.com/kubernetes-sigs/kueue/issues/2291 to eliminate the unnecessary delay. The proposed solution of incrementing `requeuingState.count` to actually exceed the `backoffLimitCount` will work both on 0.7 and 0.6.